### PR TITLE
Update UpdateProfile.js

### PR DESCRIPTION
**Pull request by Sandarbha-K**

### CODE REVIEW

File: EZPay-React/ezpay/src/components/UpdateProfile.js [hunk 1]
```diff
@@ -187,8 +256,8 @@ const UpdateProfile = () => {
             style={getInputStyle('newprofilePictureUrl')}
           />
         </div>
-        <button className="submit-button" type="submit">Update Profile</button>
```
Comment (sagarkumar3105) [inline EZPay-React/ezpay/src/components/UpdateProfile.js:190]: className="submit-button"

File: EZPay-React/ezpay/src/components/UpdateProfile.js [hunk 2]
```diff
@@ -187,8 +256,8 @@ const UpdateProfile = () => {
...
-        <button type="button" className="cancel-button" onClick={() => navigate('/view-profile')}>Cancel</button>
```
Comment (sagarkumar3105) [inline EZPay-React/ezpay/src/components/UpdateProfile.js:191]: className="canacel-button"

File: EZPay-React/ezpay/src/components/UpdateProfile.js [hunk 3]
```diff
@@ -65,116 +63,187 @@ const UpdateProfile = () => {
...
+          setErrors((prevErrors) => ({
+            ...prevErrors,
+            newMobileNumber: responseData,
+          }));
+        } else {
+          setMessage(responseData);
         }
-
-        setMessage('Profile updated successfully!');
-        setShowPasswordModal(false); // Close modal after update
```
Comment (sagarkumar3105) [inline EZPay-React/ezpay/src/components/UpdateProfile.js:118]: this one also remains

File: EZPay-React/ezpay/src/components/UpdateProfile.js [hunk 4]
```diff
@@ -65,116 +63,187 @@ const UpdateProfile = () => {
...
+        } else if (responseData.includes("Mobile number already exists")) {
+          setErrors((prevErrors) => ({
+            ...prevErrors,
+            newMobileNumber: responseData,
+          }));
+        } else {
+          setMessage(responseData);
         }
-
-        setMessage('Profile updated successfully!');
```
Comment (sagarkumar3105) [inline EZPay-React/ezpay/src/components/UpdateProfile.js:117]: this one also remains

File: EZPay-React/ezpay/src/components/UpdateProfile.js [hunk 5]
```diff
@@ -65,116 +63,187 @@ const UpdateProfile = () => {
...
+      errors.newAddress = 'Address is required';
+    }
+
+    setErrors(errors);
+    return Object.keys(errors).length === 0;
+  };
+
   const handleSubmit = (e) => {
     e.preventDefault();
-    setShowPasswordModal(true);
```
Comment (sagarkumar3105) [inline EZPay-React/ezpay/src/components/UpdateProfile.js:70]: remains

**Review by sagarkumar3105 — CHANGES_REQUESTED** [review]

**Review by Sandarbha-K — COMMENTED** [review]
Review done

**Review by sagarkumar3105 — APPROVED** [review]

**Pull request merged**